tools/ci_build/github/azure-pipelines/templates/win-ci.yml

```diff
@@ -117,23 +118,24 @@ jobs:
         arguments: '$(BuildCommand) --test --config RelWithDebInfo'
         workingFolder: "$(Build.BinariesDirectory)"
 
-    - task: VSBuild@1
-      displayName: 'Build c# RelWithDebInfo'
+    - task: MSBuild@1
+      displayName: 'Build C# RelWithDebInfo'
       inputs:
-        solution: '$(Build.SourcesDirectory)\csharp\OnnxRuntime.CSharp.sln'
-        platform: 'any cpu'
+        solution: '$(Build.SourcesDirectory)\csharp\OnnxRuntime.CSharp.proj'
+        platform: 'Any CPU'
         configuration: 'RelWithDebInfo'
         msbuildArchitecture: 'x64'
-        restoreNugetPackages: false
         workingFolder: '$(Build.SourcesDirectory)\csharp'
-        maximumCpuCount: true
 
-    - task: CmdLine@2
+    - task: MSBuild@1
       displayName: 'Test C# RelWithDebInfo'
       inputs:
-        script: |
-         FOR /F  %%f IN ('dir /s /b test\Microsoft.ML.OnnxRuntime.Tests\bin\*Tests.dll') DO %dotnetexe% vstest %%f 
-        workingDirectory: '$(Build.SourcesDirectory)\csharp'
+        solution: '$(Build.SourcesDirectory)\csharp\OnnxRuntime.CSharp.proj'
```

Comment:
One note -- this pattern does not work for x86 builds (previous pattern one does). If the old pattern is kept intact, it keeps the yaml files more consistent.

Reply:
updated the x86 pipeline